python/ray/serve/tests/unit/test_http_util.py

```diff
@@ -125,6 +125,22 @@ async def test_raises_exception(
         with pytest.raises(RuntimeError, match="oopsies"):
             await asgi_receive_proxy()
 
+    async def test_does_not_raise_key_error(
+        self, setup_receive_proxy: Tuple[ASGIReceiveProxy, MessageQueue]
+    ):
+        asgi_receive_proxy, queue = setup_receive_proxy
+
+        queue.put_nowait({"type": "foo"})
+        queue.put_nowait({"type": "bar"})
+        assert await asgi_receive_proxy() == {"type": "foo"}
+        assert await asgi_receive_proxy() == {"type": "bar"}
+
+        queue.put_nowait(KeyError("not found"))
+        _, pending = await asyncio.wait(
+            [asyncio.create_task(asgi_receive_proxy())], timeout=0.01
+        )
+        assert len(pending) == 1
```

Comment:
This is just for my own understanding. I know this test is testing KeyError not getting raised when it's put into the queue and the one below is to test some other error will be raised. How come after the `asyncio.wait` call, the task with KeyError are not getting resolved and still in the pending state? 🤔

Also, just wondering, if there is a KeyError from the user code, would we omit them the same way? (if that's possible)

Reply:
The task is to return the next `ASGI` message, if we get a `KeyError` it means the proxy is no longer handling messages for this request so we will never return another message (hence the task does not return).

There is no user code running on this path, it's all system messages between the replica and the proxy.